src/main/java/com/example/networkautomation/security/WebSecurity.java

```diff
@@ -0,0 +1,57 @@
+package com.example.networkautomation.security;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.security.web.SecurityFilterChain;
+
+
+/**
+ * This is a Javadoc comment. It provides information about the method or class.
+ * You can include a description, parameter explanations, return value, exceptions, etc.
+ */
+@Configuration
+@EnableWebSecurity
+public class WebSecurity {
+
+
+    /**
+     * This is a Javadoc comment. It provides information about the method or class.
+     * You can include a description, parameter explanations, return value, exceptions, etc.
+     */
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests((request) -> request
```

Comment:
what is happening here, explain in details as I did not get it!

Reply:
@Bean is configuring the method of that dependencies.
SecurityFilterChain -> method is used to fileter the URLs endpoints which has all the permission and which don't have. 
.authorizedHttpRequest -> this allow us to set the endpoints which is permitted and which isn't authenticated.
.formLogin -> this indicating that we are using form based login.
.logout -> this is added a logout functionality.
http.build() -> this is the final security filter which is set based on the configuration we have set.